org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMCommentFormatter.java

```diff
@@ -54,12 +54,22 @@ public void formatComment(DOMComment commentNode, XMLFormattingConstraints paren
 		}
 		int spaceStart = -1;
 		int spaceEnd = -1;
-		availableLineWidth -= 4; // count for '<-!--'
+		availableLineWidth -= 4; // count for '<!--'
+		int lineSeparatorOffset = -1;
 
 		for (int i = commentNode.getStartContent(); i < commentNode.getEndContent(); i++) {
 			char c = text.charAt(i);
 			if (Character.isWhitespace(c)) {
-				// Whitespaces
+				if (isLineSeparator(c)) {
+					// spaceStart offset not including line separator characters
+					lineSeparatorOffset = i;
```

Comment:
you should be able to set this variable outside of the if/else statement, then simplify the if/else statement